src/newButton/Button.tsx

```diff
@@ -0,0 +1,240 @@
+import { cx } from "@renderlesskit/react";
+import * as React from "react";
+import {
+  Button as ReakitButton,
+  ButtonProps as ReakitButtonProps,
+} from "reakit";
+import { useTheme } from "../theme";
+import { Spinner } from "../spinner";
+import { Dict, forwardRefWithAs } from "../utils/types";
+import { runIfFn } from "../utils";
+
+export type NonNullable<T> = Exclude<T, null | undefined>;
+
+export type ButtonProps = Omit<ReakitButtonProps, "prefix"> & {
+  /**
+   * How large should the button be?
+   *
+   * @default md
+   */
+  size?: keyof Renderlesskit.GetThemeValue<"newButton", "size">;
+  /**
+   * How the button should look?
+   *
+   * @default solid
+   */
+  variant?: keyof Renderlesskit.GetThemeValue<"newButton", "variant">;
+  /**
+   * If added, the button will only show an icon ignoring other childrens.
+   */
+  iconOnly?: React.ReactElement;
+  /**
+   * If added, the button will show an icon before the button's text.
+   */
+  suffix?: React.ReactElement;
+  /**
+   * If added, the button will show an icon before the button's text.
+   */
+  prefix?: React.ReactElement;
+  /**
+   * If `true`, the button will be disabled.
+   *
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the button will show a spinner.
+   *
+   * @default false
+   */
+  loading?: boolean;
+  /**
+   * If added, the button will show this spinner components
+   */
+  spinner?: React.ReactElement;
+};
+
+export const addIconA11y = (icon: React.ReactElement, props?: Dict) => {
+  console.log("%c props", "color: #408059", props);
+
+  return React.isValidElement(icon)
+    ? React.cloneElement(icon, {
+        // @ts-ignore
+        role: "img",
+        focusable: false,
+        "aria-hidden": true,
+        ...props,
+      })
+    : icon;
+};
+
+export const Button = forwardRefWithAs<
+  ButtonProps,
+  HTMLButtonElement,
+  "button"
+>((props, ref) => {
+  const {
+    children,
+    size = "sm",
+    variant = "solid",
+    iconOnly,
+    suffix,
+    prefix,
+    loading = false,
+    spinner,
+    disabled = false,
+    className,
+    ...rest
+  } = props;
+  const _disabled = disabled || loading;
+
+  const theme = useTheme();
+  const {
+    base,
+    size: _size,
+    variant: _variant,
+    iconOnly: _iconOnly,
+  } = theme.newButton;
+  const baseStyles = cx(
+    base,
+    !iconOnly ? _size[size] : _iconOnly.size[size],
+    _variant[variant],
+    className,
+  );
```

Comment:
Let's prefer to not have these destructured cause it can make the variable scope polluted if we are not using these anywhere else 

```ts
  const baseStyles = cx(
    theme.newButton.base,
    !iconOnly
      ? theme.newButton.size[size]
      : theme.newButton.iconOnly.size[size],
    theme.newButton.variant[variant],
    className,
  );
```

Reply:
Done & I understand.